radicle-node/src/service/session.rs

```diff
@@ -38,7 +38,9 @@ impl Default for Protocol {
 #[allow(clippy::large_enum_variant)]
 pub enum State {
     /// Initial state for outgoing connections.
-    Connecting,
+    Initial,
+    /// Connection attempted successfully.
+    Attempted,
```

Comment:
Problem with Attempted is attempted things can have failed.
What about?
 - Negotiating or
 - Attempted => Connected, Connected => Established

Reply:
The reason they are called this is to match up with the callback functions of service, which are `attempted` -> `connected`. If we rename these states, we'd also have to rename those functions, and I'd rather leave that for a different patch.